### ISSUE TITLE
Build error: unable to add AlinFoundation package

### ISSUE THREAD

**Issue by perez987** [alienator88/Sentinel#10]
Good morning. Sentinel works fine, I really like this app, very useful on Sequoia.

I've trying to compile from source, I'm on Ventura with Xcode 15.2.  What I can't do is add AlinFoundation to Xcode.

I've tried with the Sentinel source code as is, I see AlinFoundation in the left bar but Xcode doesn't find that package.

I've also tried to download the source code from AlinFoundation and add it as a dependency to the project but it shows a message that the package is corrupted or malformed.

So I can't compile Sentinel. Can you tell me something about this, how should I compile the project correctly?

Thank you very much!

**Comment by alienator88** [comment]
Hmm that is really odd.
When I add it to my projects, I select by branch name "main" instead of a version to get the latest updates. I haven't published a recent version of that swift package for a bit. Does that work for you by chance?

**Comment by perez987** [comment]
Thanks for your answer.

This is not so important since the app works perfectly. But I was curious to compile the project just for learning, to see the source code inside.

I have set by branch name "main", I think that's what you mean:

<img width="859" alt="Branch main" src="https://github.com/user-attachments/assets/3beab54d-ad86-49af-a2d0-44f5c49994b0"><br>

The errors when compiling are:

- `No such module AlinFoundation`: this error always appears
- `Package.resolved file is corrupted or malformed`: this error appears only if I set another option other than branch by name "main".

<img width="564" alt="Build errors" src="https://github.com/user-attachments/assets/0a1091cc-1b2c-4711-bca9-51e2a91d964e"><br>

If I download the AlinFoundation source code and try to add package dependencies >> Add local then the error `Package.resolved` appears.

<img width="1023" alt="Package unresolved" src="https://github.com/user-attachments/assets/eac2bbfb-168d-448e-a86b-5fbcac1fb6bb">

**Comment by alienator88** [comment]
Adding by branch is the only way it will work normally. Unless I push another release for the latest updates to the releases page.

I can't tell you why the project isn't seeing the package though. Might try cleaning the project and restarting Xcode? Not sure, sometimes Xcode acts weird.

**Comment by alienator88** [comment]
I use this package on at least 3 apps of mine and it imports fine on all. But I'm also on latest macOS and Xcode 16.

**Comment by perez987** [comment]
Thanks, I have Sequoia in another disk, I'll try and comment.

**Comment by perez987** [comment]
@alienator88 
On Sequoia and Xcode 16 the project compiles fine.

I had to remove AlinFoundation included by default in Sentinel, download the source code of AlinFoundation from GitHub and add it back to the project. After doing this, errors related to AlinFoundation have disappeared.

I have had other errors in Sentinel files that prevented compiling, when the name of a function or argument did not match those existing in Alinfoundation. E.g.:

SentinelApp.swift

`.sheet(isPresented: $updater.showSheet, content: {` -> error
`.sheet(isPresented: $updater.sheet, content: { `-> OK


UpdateSettingsView.swift

`updater.checkForUpdatesForce(showSheet: false)` -> error
`updater.checkForUpdates(force: true) `-> OK


AboutCommand.swift

`updater.checkForUpdatesForce(showSheet: false)` -> error
`updater.checkForUpdates(force: true) `-> OK

Is this correct? Or is it something wrong on my side? I'm not a coder so excuse me if I'm wrong.

At the end, all has gone fine and Sentinel compiled as it should.

**Comment by perez987** [comment]
@alienator88 
No, you don't need to grab the source code of AlinFoundation directly from the main branch.

Apparently, I had copied the project from Ventura to Sequoia and something was not right.

I downloaded the clean project again and, after fixing those 3 files that were giving errors, compilation works fine.

<img width="1006" alt="Success" src="https://github.com/user-attachments/assets/43ef3ac5-8a00-4d2d-8bb3-dd7974ce862e">
